src/server.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import unicode_literals
+import socket
+
+
+class HTTPErrors(Exception):
+    """HTTP Error exception class."""
+
+    def __init__(self, message):
+        """Intialize Error with message."""
+        self.message = message
+
+
+def parse_request(request):
+    """Parse request and if valid return URI."""
+    split_req = request.split('\r\n', 1)
+    method, uri, proto = split_req[0].split()
+    headers = split_req[1].split('\r\n\r\n')
+    split_headers = headers[0].split('\r\n')
+    header_details = [items.split(':', 1) for items in split_headers]
+    header_dict = {k.upper(): v.strip() for k, v in header_details}
+    if 'HOST' not in header_dict:
+        raise HTTPErrors('400 Host Header not found.')
```

Comment:
"Host Header not found" is not a valid HTTP response reason phrase.  The right phrase for the 400 code is "Bad Request".  Don't bend that.